src/components/ColorSwitcher.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useRef, useEffect } from "react";
+import paletteIcon from "../assets/img/palette.svg";
+
+const ColorSwitcher = ({ onColorChange }) => {
+  const [visible, setVisible] = useState(false);
+  const popoverRef = useRef();
+  const [topBarBgColor, setTopBarBgColor] = useState(
+    localStorage.getItem("topBarBgColor") || "#ffffff"
+  );
+  const [leftNavHighlightColor, setLeftNavHighlightColor] = useState(
+    localStorage.getItem("leftNavHighlightColor") || "#000000"
+  );
+  const [leftNavBgColor, setLeftNavBgColor] = useState(
+    localStorage.getItem("leftNavBgColor") || "#ffffff"
+  );
+  const [leftNavTextColor, setLeftNavTextColor] = useState(
+    localStorage.getItem("leftNavTextColor") || "#000000"
+  );
+
+  const togglePopover = () => {
+    setVisible(!visible);
+  };
+
+  const handleColorChange = (key, value) => {
+    onColorChange(key, value);
+    localStorage.setItem(key, value);
+
+    // Update state based on the key
+    switch (key) {
+      case "topBarBgColor":
+        setTopBarBgColor(value);
+        break;
+      case "leftNavHighlightColor":
+        setLeftNavHighlightColor(value);
+        break;
+      case "leftNavBgColor":
+        setLeftNavBgColor(value);
+        break;
+      case "leftNavTextColor":
+        setLeftNavTextColor(value);
+        break;
+      default:
+        break;
+    }
+  };
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (popoverRef.current && !popoverRef.current.contains(event.target)) {
+        setVisible(false);
+      }
+    };
+
+    const addIframeListeners = () => {
+      const iframes = document.getElementsByTagName("iframe");
+      Array.from(iframes).forEach((iframe) => {
+        const iframeDocument =
+          iframe.contentDocument || iframe.contentWindow.document;
+        if (iframeDocument) {
+          iframeDocument.addEventListener("mousedown", handleClickOutside);
+        }
+      });
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+
+    addIframeListeners();
+
+    const observer = new MutationObserver((mutationsList) => {
+      for (let mutation of mutationsList) {
+        if (mutation.type === "childList") {
+          addIframeListeners();
+        }
+      }
+    });
+
+    observer.observe(document.body, { childList: true, subtree: true });
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+      observer.disconnect();
+    };
+  }, []);
+
+  return (
+    <div className="relative ">
+      <img
+        src={paletteIcon}
+        onClick={togglePopover}
+        className="w-6 h-6 cursor-pointer"
+        alt="Color Palette Icon"
+      />
+      {visible && (
+        <div
```

Comment:
please use prebuilt flowbite components and restylize them as needed. it is already a dependency and is widely used throughout the UI repos. i dont see a technical reason to redo this pop-over in preference for a flowbite popover, but please keep it in mind in the future.